salt/modules/x509.py

```diff
@@ -331,10 +331,14 @@ def _parse_subject(subject):
     for nid_name, nid_num in six.iteritems(subject.nid):
         if nid_num in nids:
             continue
-        val = getattr(subject, nid_name)
-        if val:
-            ret[nid_name] = val
-            nids.append(nid_num)
+        try:
+            val = getattr(subject, nid_name)
+            if val:
+                ret[nid_name] = val
+                nids.append(nid_num)
+        except TypeError as e:
+            if e.args and e.args[0] == 'No string argument provided':
+                pass
```

Comment:
@rallytime @terminalmage @garethgreenaway Is this OK? Because while I understand why this is raised [here](https://gitlab.com/m2crypto/m2crypto/blob/master/M2Crypto/X509.py#L317) because of [this](https://gitlab.com/m2crypto/m2crypto/blob/master/M2Crypto/util.py#L68), I don't understand why one would check `e.args` on a particular bytestring and _anyway_ go mute?

I've opened a PR on this here to fix this: https://github.com/saltstack/salt/pull/46104